haystack/nodes/prompt/prompt_node.py

```diff
@@ -864,25 +865,23 @@ def run(
         prompt template.
         :param documents: The documents to be used for the prompt.
         :param meta: The meta to be used for the prompt. Usually not used.
+        :param invocation_context: The invocation context to be used for the prompt.
         """
 
-        if not meta:
-            meta = {}
         # invocation_context is a dictionary that is passed from a pipeline node to a pipeline node and can be used
         # to pass results from a pipeline node to any other downstream pipeline node.
-        if "invocation_context" not in meta:
-            meta["invocation_context"] = {}
+        invocation_context = invocation_context or {}
```

Comment:
The second implementation of Shaper needs `invocation_context` to contain a reference to all values passed down to Pipeline. Maybe that can be changed, but in order to test the two together, let's add these values to it:

```suggestion
        invocation_context = {**locals(), **{invocation_context or {}}
        invocation_context.pop("self")
```

Reply:
Can we have the new Shaper not rely on these locals? Would that be possible?